### PR TITLE
Fix AWS deployments.

### PR DESCRIPTION
**Pull request by tjholm**
Minor fixes for AWS deployments.

still TODO:
 - Need to do more normalization on AWS resource names e.g. remove characters like '.' from lambda names.

### CODE REVIEW

File: pkg/provider/pulumi/aws/gateway.go [hunk 1]
```diff
@@ -35,29 +37,68 @@ type ApiGateway struct {
...
-	b, err := args.OpenAPISpec.MarshalJSON()
+	doc := pulumi.All(nameArnPairs...).ApplyT(func(pairs []interface{}) (string, error) {
+		naps := make(map[string]string)
+
+		for _, p := range pairs {
+			if pair, ok := p.(nameArnPair); ok {
+				naps[pair.name] = pair.invokeArn
+			} else {
+				// XXX: Should not occur
+				return "", fmt.Errorf("invalid data")
```
Comment (asalkeld) [inline pkg/provider/pulumi/aws/gateway.go]: could have a more meaningful error here.. 
```suggestion
				return "", fmt.Errorf("invalid data %T %v", p, p)
```

File: pkg/stack/function_helpers.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ import (
 	"path/filepath"
 )
 
-const DefaulMembraneVersion = "v0.12.1-rc.5"
+const DefaultMembraneVersion = "v0.13.0-rc.12"
```
Comment (asalkeld) [inline pkg/stack/function_helpers.go:25]: our "make generate" should automatically update this from our go.sum :thinking:

**Review by asalkeld — APPROVED** [review]
well done! minor nits if you have time

File: pkg/provider/pulumi/aws/gateway.go (hunk 1 above)
Comment (tjholm) [reply pkg/provider/pulumi/aws/gateway.go]: done

File: pkg/stack/function_helpers.go (hunk 2 above)
Comment (tjholm) [reply pkg/stack/function_helpers.go:25]: Good idea, I've been doing a bit more reading on doing this with go generate. Would the idea in this case be that we could create a seperate gen script for individual packages with a generate directive to create that?

In this specific case would we generate a new file e.g. pkg/stack/const.go with this content but have the generate directive in the source file that depends on this constant?

Comment (asalkeld) [reply pkg/stack/function_helpers.go:25]: https://github.com/nitrictech/newcli/pull/48

**Pull request merged**